src/Lykke.Job.TradeDataAggregator.Services/ShutdownManager.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Threading.Tasks;
+using Common.Log;
+using Lykke.Job.TradeDataAggregator.Core.Services;
+
+namespace Lykke.Job.TradeDataAggregator.Services
+{
+    // NOTE: Sometimes, shutdown process should be expressed explicitly. 
+    // If this is your case, use this class to manage shutdown.
+    // For example, sometimes some state should be saved only after all incoming message processing and 
+    // all periodical handler was stopped, and so on.
+    
+    public class ShutdownManager : IShutdownManager
+    {
+        private readonly ILog _log;
+
+        public ShutdownManager(ILog log)
+        {
+            _log = log;
+        }
+
+        public async Task StopAsync()
+        {
+            // TODO: Implement your shutdown logic here. Good idea is to log every step
```

Comment:
ScanClientsHandler should be stopped here

Reply:
Yep